src/kakarot/interpreter.cairo

```diff
@@ -1029,8 +1040,16 @@ namespace Internals {
             return evm;
         }
 
+        // Write bytecode and cache the final code valid jumpdests to Account
         let account = State.get_account(evm.message.address.evm);
         let account = Account.set_code(account, evm.return_data_len, evm.return_data);
+        let account = Account.set_created(account, TRUE);
+
+        let (valid_jumpdests_start, valid_jumpdests) = Helpers.initialize_jumpdests(
+            account.code_len, account.code
+        );
+        let account = Account.set_valid_jumpdests(account, valid_jumpdests_start, valid_jumpdests);
+
```

Comment:
we may want to just create a `new Account` with all the parameters set to their final value, instead of using the methods and creating 5x a new account